src/p2p/net_node.inl

```diff
@@ -1789,4 +1876,18 @@ namespace nodetool
 
     return true;
   }
+
+  template<class t_payload_net_handler>
+  bool node_server<t_payload_net_handler>::remove_outgoing_connection_from_anchor_peerlist(p2p_connection_context& context)
+  {
+    if (!m_net_server.is_stop_signal_sent() && !context.m_is_income) {
```

Comment:
I think this is confusing, and the double test should be in the caller. Subjective though.